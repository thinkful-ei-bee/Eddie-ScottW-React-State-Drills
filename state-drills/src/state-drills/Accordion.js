import React from 'react';

class Accordion extends React.Component{


  static defaultProps = {
    sections: [],
  }

  state = {
   currentSection: null,
  }

  handleClick = (index) => {

    if (index === this.state.currentSection) {
      this.setState({
        currentSection: null,
      });
    }
    else {
      this.setState({
        currentSection: index,
      });
    }
  }

  render(){

    const buttons = this.props.sections.map((section, index) => (

      <li key={index}>
      <button onClick={() => this.handleClick(index)}>
      {section.title}
      </button>
      {
      this.state.currentSection === index &&
      <p>{section.content}</p>
      }
      </li>

    ))

    return (
      <ul>
        {buttons}
      </ul>
    )
  }
}

export default Accordion;