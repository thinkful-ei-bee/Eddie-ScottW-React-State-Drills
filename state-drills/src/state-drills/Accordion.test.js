import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe(`Accordion Component`, () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders an empty li when sections prop is not given', ()=>{
    const tree = renderer.create(<Accordion />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders no sections as active by default',() => {
    const tree = renderer.create(<Accordion sections={sections}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

})