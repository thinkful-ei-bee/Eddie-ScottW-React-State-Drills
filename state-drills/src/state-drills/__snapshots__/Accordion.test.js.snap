// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component renders an empty li when sections prop is not given 1`] = `<ul />`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
